### PR TITLE
Diff module added

### PR DESCRIPTION
**Pull request by nyzd**
Diff module added The new Diff struct finds the deference between two Vecs. It's almost completed and only the Delete Action remains. This new module is generic compatible and can be used with any type of data.

### CODE REVIEW

**Comment by Al-Abd** [comment]
I suggest we use less acronyms, and please write more comments in the code.

File: src/diff.rs
Comment (al-abd) [inline src/diff.rs:1]: I suggest we use less acronyms, and please write more comments in the code.

File: src/diff.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                None => result.push(Difference::Insert(object)),
+            }
+        }
+
+        result
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (al-abd) [inline src/diff.rs:58]: Shouldn't we leave the test codes in test files?

Explain the reason for this.

**Review by al-abd — CHANGES_REQUESTED** [review]
Already approved, just answer comments.

File: src/diff.rs (hunk 1 above)
Comment (nyzd) [reply src/diff.rs:58]: Currently we only have separate test files for routers, I think creating file for every module for only fixed amounts of tests would be annoying. And over complicated.

File: src/diff.rs
Comment (nyzd) [reply src/diff.rs:1]: I already said in the description of this pr that the Diff module is not complete yet. With that said I will make sure the diff module is readable and efficient.

**Pull request merged**